crates/autopilot/src/solvable_orders.rs

```diff
@@ -360,7 +361,15 @@ impl SolvableOrdersCache {
             None => self.persistence.all_solvable_orders(min_valid_to).boxed(),
         };
 
-        fetch_orders.await
+        let mut orders = fetch_orders.await?;
+
+        // Move the checkpoint slightly back in time to mitigate race conditions
+        // caused by inconsistencies of stored timestamps. See #2959 for more details.
+        // This will cause us to fetch orders created or cancelled in the buffer
+        // period multiple times but that is a small price to pay for not missing
+        // orders.
+        orders.fetched_from_db -= chrono::TimeDelta::seconds(60);
```

Comment:
Detaching this logic from the actual value creation might lead to errors in the future.

Reply:
It was either this or already decreasing the value in 2 different places in the infra code.
Ultimately this seemed like a domain logic decision to me so I put it here.